crates/task-impls/src/vid.rs

```diff
@@ -355,6 +364,53 @@ where
                     ))
                     .await;
             }
+
+            HotShotEvent::TransactionsSequenced(payload, _metadata, view) => {
+                // encode transactions for disperse calculation
+                let encoded_txns = match payload.encode() {
+                    Ok(encoded) => encoded,
+                    Err(e) => {
+                        error!("Failed to encode the block payload: {:?}.", e);
+                        return None;
+                    }
+                };
+
+                // TODO <https://github.com/EspressoSystems/HotShot/issues/1686>
+                let srs = test_srs(NUM_STORAGE_NODES);
+                // TODO We are using constant numbers for now, but they will change as the quorum size
+                // changes.
+                // TODO <https://github.com/EspressoSystems/HotShot/issues/1693>
+                let vid = VidScheme::new(NUM_CHUNKS, NUM_STORAGE_NODES, &srs).unwrap();
+                let vid_disperse = vid
+                    .disperse(encoded_txns.into_iter().collect::<Vec<u8>>())
+                    .unwrap();
+
+                // TODO never clone a block
+                // https://github.com/EspressoSystems/HotShot/issues/1858
+                self.event_stream
+                    .publish(HotShotEvent::BlockReady(
+                        Proposal {
+                            data: VidDisperse {
+                                view_number: view,
+                                payload_commitment: payload.commit(),
```

Comment:
Not sure whether this comment is with scope of this PR but...

I think this should be `vid_disperse.commit` instead of `payload.commit()`. That means we need to change the `VidDisperse` struct so that the type of `payload_commitment` is `<VidScheme as VidSchemeTrait>::Commit` just like the other fields `shares`, `common`.

The goal is to remove `Committable` from `BlockPayload`, correct? In that case we should never be calling `commit()`.

Reply:
Yeah after removing `Committable` from `BlockPayload`, calling `commit()` will no longer work.

Reply:
But if it's too complicated to change this, you could leave it to me. I'm going to update other `commit()` calls anyway.

Reply:
Okay, I will leave it to your changes (removing commit)